presto-main/src/main/java/io/prestosql/security/ConnectorAccessControlForProcedures.java

```diff
@@ -0,0 +1,354 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.security;
+
+import io.prestosql.metadata.QualifiedObjectName;
+import io.prestosql.spi.connector.CatalogSchemaName;
+import io.prestosql.spi.connector.CatalogSchemaTableName;
+import io.prestosql.spi.connector.ColumnMetadata;
+import io.prestosql.spi.connector.ConnectorAccessControl;
+import io.prestosql.spi.connector.ConnectorSecurityContext;
+import io.prestosql.spi.connector.SchemaRoutineName;
+import io.prestosql.spi.connector.SchemaTableName;
+import io.prestosql.spi.security.PrestoPrincipal;
+import io.prestosql.spi.security.Privilege;
+import io.prestosql.spi.security.ViewExpression;
+import io.prestosql.spi.type.Type;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.Iterables.getFirst;
+import static java.util.Objects.requireNonNull;
+
+public class ConnectorAccessControlForProcedures
+        implements ConnectorAccessControl
+{
+    private final AccessControl accessControl;
+    private final SecurityContext securityContext;
+    private final String catalogName;
+
+    public ConnectorAccessControlForProcedures(AccessControl accessControl, SecurityContext securityContext, String catalogName)
+    {
+        this.accessControl = requireNonNull(accessControl, "accessControl is null");
+        this.securityContext = requireNonNull(securityContext, "securityContext is null");
+        this.catalogName = requireNonNull(catalogName, "catalogName is null");
+    }
+
+    @Override
+    public void checkCanCreateSchema(ConnectorSecurityContext context, String schemaName)
+    {
+        checkArgument(context == null, "connectorSecurityContext should be null");
```

Comment:
I would say "must" rather than "should" since this is a requirement (and "should" implies a recommendation)